File: 299. Bulls and Cows/_299.py
class Solution:
    def getHint(self, secret: str, guess: str) -> str:
        mydict = {}
        bull = 0
        cows = 0

        for i in secret:
            v = mydict.get(i, 0)
            mydict[i] = v+1

        for i in guess:
            v = mydict.get(i, 0)
            if v != 0:
                cows+=1
                mydict[i] = v-1
        
        for i in range(len(secret)):
            if secret[i] == guess[i]:
                cows -= 1
                bull += 1

        return str(bull)+'A'+str(cows)+'B'

    # 这个用库的方式还是要多学习，特别是collections这个库
    def getHint1(self, secret: str, guess: str) -> str:
        from collections import Counter
        s, g = Counter(secret), Counter(guess)
        a = sum(i == j for i, j in zip(secret, guess))
        return '%sA%sB' % (a, sum((s & g).values()) - a)

File: 299. Bulls and Cows/299.go
package leetcode

import(
	"strconv"
)

func getHint(secret string, guess string) string {
	mydict := make(map[rune]int)
	cows := 0
	bull := 0

	for _, v := range secret{
		dictv, ok := mydict[v]
		if ok{
			mydict[v] = dictv + 1
		}else{
			mydict[v] = 1
		}
	}

	for i, v := range guess{
		dictv, ok := mydict[v]
		if ok && dictv>0{
			cows++
			mydict[v] = dictv-1
		}
		if v == rune(secret[i]){
			cows--
			bull++
		}
	}
	return strconv.Itoa(bull)+"A"+strconv.Itoa(cows)+"B"
}